graylog2-web-interface/src/components/grok-patterns/GrokPatternInput.jsx

```diff
@@ -0,0 +1,137 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { Input } from 'components/bootstrap';
+import { Row, Col, Button, ListGroup, ListGroupItem } from 'react-bootstrap';
+
+import GrokPatternInputStyle from './GrokPatternInput.css';
+
+class GrokPatternInput extends React.Component {
+  static propTypes = {
+    pattern: PropTypes.string,
+    patterns: PropTypes.array,
+    onPatternChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    pattern: '',
+    patterns: [],
+    onPatternChange: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      patternFilter: '',
+      activeListItem: -1,
+    };
+  }
+
+  shownListItems = [];
+
+  _onPatternChange = (e) => {
+    this.props.onPatternChange(e.target.value);
+  };
+
+  _onPatternFilterChange = (e) => {
+    this.setState({ patternFilter: e.target.value, activeListItem: -1 });
+  };
+
+  _onPatternFilterKeyDown = (e) => {
+    const ARROW_DOWN = 40;
+    const ARROW_UP = 38;
+    const ENTER = 13;
+
+    let activeListItem = 0;
+    let domElement;
+    switch (e.keyCode) {
+      case ARROW_DOWN:
+        activeListItem = this.state.activeListItem + 1;
+        if (activeListItem >= this.shownListItems.length) {
+          return;
+        }
+        domElement = document.getElementById(`list-item-${activeListItem}`);
+        domElement.scrollIntoView({ behavior: 'smooth' });
+        this.setState({ activeListItem: activeListItem });
+        e.preventDefault();
+        break;
+      case ARROW_UP:
+        activeListItem = this.state.activeListItem - 1;
+        if (activeListItem < 0) {
+          return;
+        }
+        domElement = document.getElementById(`list-item-${activeListItem}`);
+        domElement.scrollIntoView();
+        this.setState({ activeListItem: activeListItem });
+        e.preventDefault();
+        break;
+      case ENTER:
```

Comment:
This does not properly work when pressing enter while the focus is in the filter, example:

- Go to the input filter
- Type anything (optional)
- Press enter
- `%{undefined}` is added to the list